hr_expense_invoice/models/hr_expense.py

```diff
@@ -88,7 +63,7 @@ def _prepare_own_account_transfer_move_vals(self):
             ],
             limit=1,
         )
-        employee_partner = self.employee_id.sudo().address_home_id.commercial_partner_id
+        employee_partner = self.employee_id.sudo().work_contact_id
```

Comment:
Although I understand the reason for this change (as the private address has been removed), this is not truly correct, as the work contact is not really representing the employee, and you can't put a journal entry linked to a children contact. How Odoo handles this in other parts where the debt is to the employee?

Reply:
Actually I took it from Odoo.

Example 1:
[v17](https://github.com/odoo/odoo/blob/17.0/addons/hr_expense/models/hr_expense.py#L879)
[v16](https://github.com/odoo/odoo/blob/16.0/addons/hr_expense/models/hr_expense.py#L690)

Example 2:
[v17](https://github.com/odoo/odoo/blob/17.0/addons/hr_expense/models/hr_expense_sheet.py#L715)
[v16](https://github.com/odoo/odoo/blob/16.0/addons/hr_expense/models/hr_expense.py#L1325)